src/core/psi_container.jl

```diff
@@ -163,7 +163,7 @@ function psi_container_init!(
     end
 
     if get_use_forecast_data(settings)
-        total_number_of_devices = length(PSY.get_components(PSY.Device, sys))
+        total_number_of_devices = length(get_available_components(PSY.Device, sys))
```

Comment:
Seems like you never want to do anything with components that are not available.  Should we have a flag in PowerSystems that prevents those components from being added?  Maybe this is not appropriate...just an idea to simplify the logic.

Reply:
I rather leave it as it is and have the users make modifications to their systems if they want to make a device unavailable explicitly.